core/shell.py

```diff
@@ -378,6 +380,15 @@ def Main(lang, arg_r, environ, login_shell, loader, readline):
 
   dir_stack = state.DirStack()
 
+  # Find common directories, these are consumed by some builtins and then much
+  # later used while setting up an interactive shell.
+  home_dir = pyos.GetMyHomeDir()
+  assert home_dir is not None
+
+  # init the HISTFILE variable to our default history file
+  history_filename = os_path.join(home_dir, '.config/oil/history_%s' % lang)
+  mem.SetValue(location.LName('HISTFILE'), value.Str(history_filename), scope_e.GlobalOnly)
```

Comment:
There's a helper `state.SetGlobalString('HISTFILE', history_filename)` you can use here